### ISSUE TITLE
[emacs] provide hint that suggestion limit has been reached

### ISSUE THREAD

**Issue by kermorgant** [phpactor/phpactor#636]
Hi,

Currently, emacs' completion backend caches the sugestions from phpactor and dot not trigger any extra ``complete`` action unless there's any change in the prefix or the list of suggestions gets empty.

I've opened this [pr](https://github.com/emacs-php/phpactor.el/pull/53) that lets the user configure the suggestion limit (should be aligned on phpactor's setting), resulting in a smarter uise of caching.

But if a user starts configuring this setting on a per project basis, he will have to do the same with this emacs setting and it starts to be cumbersome.

It would be simpler on the emacs side to have a flag in the suggestions, telling that the limit has been reached (the ``company-phpactor--completion-limit`` could be removed).

**Comment by kermorgant** [comment]
Determining that the limit has been reached could also remain delagated to the client by sending sending the value of the limit would in the rpc response.

**Comment by dantleech** [comment]
Currently the limit is applied only to class completions, so wouldn't make much sense to return the completion limit without refactoring to make this limit apply to the entire suggestion set.

Some more options:

1. Make the completion limit apply to the whole suggestion set (trivial, but would also need to refactor to allow results to be interpolated so that one Completor doesn't fill the quota all by itself). Add the limit information to the response (could be implemented as pagination).
2. Allowing the client to specify the limit in the RPC request.

Doing of both of these things would probably make sense, but would `2` solve your problem?

**Comment by kermorgant** [comment]
>2. Allowing the client to specify the limit in the RPC request

Yes, that would work. Would you remove the current config item from phpactor then ?
